App/Sagas/NodeLifecycle.ts

```diff
@@ -95,10 +100,19 @@ export function * handleCreateNodeRequest (dispatch: Dispatch) {
 function * createAndStartNode(dispatch: Dispatch): any {
   try {
     yield put(TextileNodeActions.creatingNode())
+    const repoPathExists: boolean = yield call(RNFS.exists, REPO_PATH)
+    if (!repoPathExists) {
+      yield call(RNFS.mkdir, REPO_PATH)
+      yield call(moveTextileFiles)
+    }
     yield call(newTextile, REPO_PATH, LOG_LEVELS)
     yield put(TextileNodeActions.createNodeSuccess())
     yield put(TextileNodeActions.startingNode())
     yield call(start)
+    const sessions: ReadonlyArray<CafeSession> = yield call(cafeSessions)
+    if (sessions.length < 1) {
+      yield call(discoverAndRegisterCafes)
```

Comment:
If there are no sessions on record, discover and register cafes. This happens as part of the node create and start sequence, so the node isn't considered 'started' until we are able to register cafes. This means, the UI is blocked until we register cafes. If we can't register cafes for any reason, that's considered a show stopper and we show the error to the user, but they can't proceed.